app/controllers/katello/api/v2/repositories_controller.rb

```diff
@@ -448,6 +462,20 @@ def repository_params
       keys += [{:docker_tags_whitelist => []}, :docker_upstream_name] if params[:action] == 'create' || @repository&.docker?
       keys += [:ansible_collection_requirements, :ansible_collection_auth_url, :ansible_collection_auth_token] if params[:action] == 'create' || @repository&.ansible_collection?
       keys += [:label, :content_type] if params[:action] == "create"
+
+      if params[:action] == 'create' || @repository&.generic?
+        RepositoryTypeManager.repository_types.each do |_, type|
+          type.generic_remote_options.each do |option|
+            if option.type == Array
+              keys += [{option.name => []}]
+            elsif option.type == Hash
+              keys += [{option.name => {}}]
+            else
+              keys += [option.name]
+            end
+          end
+        end
+      end
```

Comment:
we might wanna move this to a new method that gets called from here (would probably fix the rubocop failure without disabling the cop)

Also i think we also want this for update?

Reply:
Yeah, @rverdile it looks like PUT at /katello/api/repositories/:id isn't working because of it.